gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDatasetFinder.java

```diff
@@ -49,37 +54,63 @@
   public static final String TABLE_PATTERN_KEY = HIVE_DATASET_PREFIX + ".table.pattern";
   public static final String DEFAULT_TABLE_PATTERN = "*";
 
-  private final HiveRegProps hiveProps;
   private final Properties properties;
   private final HiveMetastoreClientPool clientPool;
   private final FileSystem fs;
-  private final String db;
-  private final String tablePattern;
+  private final WhitelistBlacklist whitelistBlacklist;
 
   public HiveDatasetFinder(FileSystem fs, Properties properties) throws IOException {
+    this(fs, properties, createClientPool(properties));
+  }
 
-    Preconditions.checkArgument(properties.containsKey(DB_KEY));
-
+  protected HiveDatasetFinder(FileSystem fs, Properties properties, HiveMetastoreClientPool clientPool) throws IOException {
+    this.properties = properties;
+    this.clientPool = clientPool;
     this.fs = fs;
-    this.clientPool = HiveMetastoreClientPool.get(properties,
-        Optional.fromNullable(properties.getProperty(HIVE_METASTORE_URI_KEY)));
-    this.hiveProps = this.clientPool.getHiveRegProps();
 
-    this.db = properties.getProperty(DB_KEY);
-    this.tablePattern = properties.getProperty(TABLE_PATTERN_KEY, DEFAULT_TABLE_PATTERN);
-    this.properties = properties;
+    String whitelistKey = HIVE_DATASET_PREFIX + "." + WhitelistBlacklist.WHITELIST;
+    Preconditions.checkArgument(properties.containsKey(DB_KEY) || properties.containsKey(whitelistKey),
+        String.format("Must specify %s or %s.", DB_KEY, whitelistKey));
+
+    Config config = ConfigFactory.parseProperties(properties);
+
+    if (properties.containsKey(DB_KEY)) {
+      this.whitelistBlacklist = new WhitelistBlacklist(this.properties.getProperty(DB_KEY) + "."
```

Comment:
It is. This block is for backwards compatibility. However, the intended use is to set `hive.dataset.whitelist` and `hive.dataset.blacklist` which is taken care of by the else block. (See unit test)